cmd/mdatagen/templates/metrics.go.tmpl

```diff
@@ -216,27 +231,38 @@ func WithStartTime(startTime pcommon.Timestamp) metricBuilderOption {
 		mb.startTime = startTime
 	}
 }
-// WithResourceAttributesSettings sets ResourceAttributeSettings on the metrics builder.
-func WithResourceAttributesSettings(ras ResourceAttributesSettings) metricBuilderOption {
-    return func(mb *MetricsBuilder) {
-        mb.resourceAttributesSettings = ras
-    }
+
+func DefaultMetricsBuilderConfig() MetricsBuilderConfig {
+   return MetricsBuilderConfig {
+        Metrics: DefaultMetricsSettings(),
+        ResourceAttributes: DefaultResourceAttributesSettings(),
+   }
+}
+
+func (mbc MetricsBuilderConfig) WithMetrics(ms MetricsSettings) MetricsBuilderConfig {
+    mbc.Metrics = ms
+    return mbc
+}
+
+func (mbc MetricsBuilderConfig) WithResourceAttributes(ras ResourceAttributesSettings) MetricsBuilderConfig {
+    mbc.ResourceAttributes = ras
+    return mbc
```

Comment:
Why is this needed if you pass `MetricsBuilderConfig` to the constructor

Reply:
Not sure I understand how can it reduce the amount of code changes. you just need to replace the first argument in all the `NewMetricsBuilder` calls instead of introducing and using these methods